components/provisioning/org.wso2.carbon.identity.provisioning/src/test/java/org/wso2/carbon/identity/provisioning/ProvisioningUtilTest.java

```diff
@@ -0,0 +1,385 @@
+
+        when(mockClaimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(inboundClaimDialect,
+                keySet, tenantDomain, false)).thenReturn(claimMappings);
+        Map<ClaimMapping, List<String>> result = null;
+        try {
+            result = ProvisioningUtil.getMappedClaims(outboundClaimDialect,
+                    claimValueMap, inboundClaimMappingDialect, (Map<ClaimMapping,
+                            List<String>>)
+                            outboundClaimValueMappings, tenantDomain);
+            assertNotNull(result);
+
+        } catch (IdentityApplicationManagementException e) {
+
+        }
+    }
+
+    @Test(dataProvider = "getInboundClaimData2")
+    public void testGetMappedClaims(Object outboundClaimMappings, Object inboundClaimValueMap,
+                                    Object inboundClaimMappings,
+                                    Object outboundClaimValueMappings) {
+
+        Map<ClaimMapping, List<String>> result = null;
+        try {
+            result = ProvisioningUtil.getMappedClaims((ClaimMapping[]) outboundClaimMappings,
+                    (Map<String, String>) inboundClaimValueMap, (ClaimMapping[]) inboundClaimMappings,
+                    (Map<ClaimMapping,
+                            List<String>>) outboundClaimValueMappings);
+            assertNotNull(result);
+
+        } catch (IdentityApplicationManagementException e) {
+            //
+        }
+    }
+
+
+    @Test(dataProvider = "getInboundClaimData3")
+    public void testGetMappedClaims(Object outboundClaimMappings, Object inboundClaimValueMap,
+                                    String inboundClaimMappingDialect,
+                                    Object outboundClaimValueMappings, String tenantDomain) throws Exception {
+
+        Map claimValueMap = (Map<String, String>) inboundClaimValueMap;
+        Set keySet = null;
+        if (MapUtils.isNotEmpty(claimValueMap)) {
+            keySet = claimValueMap.keySet();
+        }
+
+        Map<String, String> claimMappings = new HashMap<>();
+
+        claimMappings.put(localClaimUri, "testOutboundUser");
+        claimMappings.put(localClaimUri + "1", "testOutboundUser1");
+        claimMappings.put("test", localClaimUri);
+        claimMappings.put(localClaimUri, localClaimUri);
+
+        when(mockClaimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(inboundClaimDialect,
+                keySet, tenantDomain, true)).thenReturn(claimMappings);
+
+        Map<ClaimMapping, List<String>> result = null;
+        try {
+            result = ProvisioningUtil.getMappedClaims((ClaimMapping[]) outboundClaimMappings, claimValueMap,
+                    inboundClaimMappingDialect,
+                    (Map<ClaimMapping, List<String>>) outboundClaimValueMappings, tenantDomain);
+            assertNotNull(result);
+
+        } catch (Exception e) {
+            assertTrue(e instanceof IdentityApplicationManagementException);
```

Comment:
If you are expecting this exception declare this as an expected exception as @Test (expectedExceptions = IdentityApplicationManagementException.class)